### PR TITLE
Document tag-based filtering with package_step_product_search

### PR DESCRIPTION
**Pull request by noon-ng**

### CODE REVIEW

File: docs/reference/tags/package_step_product_search_tag/index.md [hunk 1]
```diff
@@ -64,10 +64,35 @@ The tag returns an `items` array and a `Paginate` pagination object to the block
...
+
+Some examples of these might be "Star Rating", "Amenities", "Distance" etc. Consult the company for the available tags and categories.
+
+Filtering is aware of `include_organisation_products` and will include the organisation's products if the flag is set.
+
+**Syntax**
+
+`package_step_product_search` filters results if one or more `search` query parameters are present in the URL. The search query parameters are in the format `search[<tag category>][]=<tag value>`; add as many as needed to filter the results.
+
+Multiple categories and values by repeating the `search[][]` query parameter.
```
Comment (CatalinaTudor) [inline docs/reference/tags/package_step_product_search_tag/index.md:85]: **nit** This feels a bit cryptic. Should we say something like: "You can specify multiple categories and values by repeating the `search[][]` query parameter.`?

**Review by CatalinaTudor — APPROVED** [review]
Looks good to me 👍 

Probably not one for now but maybe we should consider exposing the search tag at this level as well next to the other search options:

File: docs/reference/tags/package_step_product_search_tag/index.md [hunk 2]
```diff
@@ -64,10 +64,35 @@ The tag returns an `items` array and a `Paginate` pagination object to the block
...
 
+## Tag-based filtering
+
+If the company has added tags to the step's products, these can be used to filter the results outputted by the search tag.
+
+Product tags are textual values that identify attributes of the products. They are grouped under categories which the company is also able to define.
+
+Some examples of these might be "Star Rating", "Amenities", "Distance" etc. Consult the company for the available tags and categories.
+
+Filtering is aware of `include_organisation_products` and will include the organisation's products if the flag is set.
```
[the remark this answers is not in the document]
Comment (noon-ng) [reply docs/reference/tags/package_step_product_search_tag/index.md:79]: Thanks! Hadn't seen it, will update

File: docs/reference/tags/package_step_product_search_tag/index.md [hunk 3]
```diff
@@ -64,9 +64,28 @@ The tag returns an `items` array and a `Paginate` pagination object to the block
 
 ## Search Params
 
-#### include_organisation_products
```
Comment (CatalinaTudor) [inline docs/reference/tags/package_step_product_search_tag/index.md:67]: This is still used in the tag itself, right: https://github.com/easolhq/easol/blob/main/app/tags/package_step_product_search_tag.rb#L32? it's just its effect is different now 🤔

**Pull request merged**